great_expectations/experimental/datasources/spark_azure_blob_storage_datasource.py

```diff
@@ -0,0 +1,167 @@
+                        f'Due to exception: "{str(e)}", "azure_client" could not be created.'
+                    ) from e
+            else:
+                raise SparkAzureBlobStorageDatasourceError(
+                    'Unable to create "SparkAzureBlobStorageDatasource" due to missing azure.storage.blob dependency.'
+                )
+
+            self._azure_client = azure_client
+
+        return azure_client
+
+    def test_connection(self, test_assets: bool = True) -> None:
+        """Test the connection for the SparkAzureBlobStorageDatasource.
+
+        Args:
+            test_assets: If assets have been passed to the SparkAzureBlobStorageDatasource, whether to test them as well.
+
+        Raises:
+            TestConnectionError: If the connection test fails.
+        """
+        if self._azure_client is None:
+            raise TestConnectionError(
+                "Unable to load azure.storage.blob.BlobServiceClient (it is required for SparkAzureBlobStorageDatasource)."
+            )
+
+        if self.assets and test_assets:
+            for asset in self.assets.values():
+                asset.test_connection()
+
+    def add_csv_asset(
+        self,
+        name: str,
+        batching_regex: Union[re.Pattern, str],
+        container: str,
+        name_starts_with: str = "",
+        delimiter: str = "/",
+        order_by: Optional[BatchSortersDefinition] = None,
+    ) -> CSVAsset:
+        """Adds a CSV DataAsst to the present "SparkAzureBlobStorageDatasource" object.
+
+        Args:
+            name: The name of the CSV asset
+            batching_regex: regex pattern that matches csv filenames that is used to label the batches
+            container: container name for Microsoft Azure Blob Storage
+            name_starts_with: Microsoft Azure Blob Storage object name prefix
+            delimiter: Microsoft Azure Blob Storage object name delimiter
+            order_by: sorting directive via either list[BatchSorter] or "{+|-}key" syntax: +/- (a/de)scending; + default
+        """
+        batching_regex_pattern: re.Pattern = self.parse_batching_regex_string(
+            batching_regex=batching_regex
+        )
+        order_by_sorters: list[BatchSorter] = self.parse_order_by_sorters(
+            order_by=order_by
+        )
+
+        asset = CSVAsset(
+            name=name,
+            batching_regex=batching_regex_pattern,
+            order_by=order_by_sorters,
+        )
+
+        data_connector: DataConnector = AzureBlobStorageDataConnector(
+            datasource_name=self.name,
+            data_asset_name=name,
+            batching_regex=batching_regex_pattern,
+            azure_client=self._azure_client,
+            account_name=self._account_name,
+            container=container,
+            name_starts_with=name_starts_with,
+            delimiter=delimiter,
+            file_path_template_map_fn=AzureUrl.AZURE_BLOB_STORAGE_HTTPS_URL_TEMPLATE.format,
+        )
```

Comment:
@Kilo59 this is exactly the focus of my next pull request (mentioned in Slack).  Thanks.

Reply:
@Kilo59 and @alexsherstinsky believe that a private `Datasource_get_data_connector(self, <common_args>, **kwargs)`  interface method is the best trade-off (and in this case `kwargs` is acceptable, since it is a private interface method, and the subclasses indeed use different arguments, depending on the storage environment).  Thanks. /cc @donaldheppner